File: apps/android/app/src/main/java/com/notiving/notiving/shell/router/DeepLinkHandler.kt
package com.notiving.notiving.shell.router

import android.content.Intent
import android.net.Uri

object DeepLinkHandler {
    fun handle(intent: Intent, router: ShellRouter) {
        val uri = intent.data ?: return
        handle(uri, router)
    }

    fun handle(uri: Uri, router: ShellRouter) {
        val path = when (uri.scheme) {
            "notiving" -> "/${uri.host.orEmpty()}${uri.path.orEmpty()}"
            "https", "http" -> {
                if (uri.host != "notiving.com") return
                uri.path?.ifEmpty { "/" } ?: "/"
            }
            else -> return
        }

        if (path.startsWith("/oauth/callback")) {
            val token = uri.getQueryParameter("token") ?: return
            // TODO: persist token via SessionManager when available
            router.navigate("/profile")
            return
        }

        router.navigate(path)
    }
}

File: apps/ios/notiving/notiving/Shell/Router/DeepLinkHandler.swift
import Foundation

struct DeepLinkHandler {
    static func handle(_ url: URL, router: ShellRouter = .shared, session: SessionManager = .shared) {
        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false) else { return }

        let path: String
        if url.scheme == "notiving" {
            path = "/\(url.host ?? "")\(url.path)"
        } else if url.host == "notiving.com" || url.host == "h5.notiving.com" {
            path = url.path.isEmpty ? "/" : url.path
        } else {
            return
        }

        if path.hasPrefix("/oauth/callback") {
            if let token = components.queryItems?.first(where: { $0.name == "token" })?.value {
                session.accessToken = token
                router.navigate(to: "/profile")
            }
            return
        }

        router.navigate(to: path)
    }
}
